storm-core/src/jvm/org/apache/storm/daemon/supervisor/RunAsUserContainer.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.daemon.supervisor;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.storm.container.ResourceIsolationInterface;
+import org.apache.storm.generated.LocalAssignment;
+import org.apache.storm.utils.LocalState;
+import org.apache.storm.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RunAsUserContainer extends BasicContainer {
+    private static final Logger LOG = LoggerFactory.getLogger(RunAsUserContainer.class);
+
+    public RunAsUserContainer(int port, LocalAssignment assignment, Map<String, Object> conf, String supervisorId,
+            LocalState state, ResourceIsolationInterface resourceIsolationManager, boolean recover) throws IOException {
+        super(port, assignment, conf, supervisorId, state, resourceIsolationManager, recover);
+        if (Utils.isOnWindows()) {
+            throw new RuntimeException("ERROR: Windows doesn't support running workers as different users yet");
+        }
+    }
+
+    public RunAsUserContainer(String workerId, Map<String, Object> conf, String supervisorId,
+            ResourceIsolationInterface resourceIsolationManager) throws IOException {
+        super(workerId, conf, supervisorId, resourceIsolationManager);
+        if (Utils.isOnWindows()) {
+            throw new RuntimeException("ERROR: Windows doesn't support running workers as different users yet");
```

Comment:
UnsupportedOperationException, and in fact we could add a Utils#unsupportedOnWindows method and just call it as we would a validator.